integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -620,6 +609,9 @@ object CarbonDataRDDFactory {
             carbonLoadModel)
         OperationListenerBus.getInstance()
           .fireEvent(loadTablePreStatusUpdateEvent, operationContext)
+        val segmentFileName =
```

Comment:
mergeIndex call will be writing the segment file and updating the table status file right? why do we need here again?
are you calling just to update the `segmentMetaDataInfo`?
I think here, we need only if the merge index is false, please check once and confirm